ush/ufsda/yamltools.py

```diff
@@ -1,9 +1,48 @@
 import datetime
 import os
 import re
+import logging
 from solo.yaml_file import YAMLFile
 from solo.template import TemplateConstants, Template
 from ufsda.misc_utils import isTrue
+import pygw
+
+logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s',
+                    level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
+
+
+class DAYAMLFile(pygw.yaml_file.YAMLFile):
```

Comment:
if you are going to go this route, I suggest a "cleanup" function that removes the YAML pieces if the file doesn't exist that runs on an existing YAML file after it is created. However this is not what I would suggest as a permanent solution.

I understand this workaround is due to using R2D2... but we aren't going to be using R2D2 in GFS/GDAS at least not initially. So should we be moving away from R2D2 sooner rather than later. Should this be the catalyst? 

I would much rather create the `obs_list` on the fly based on available observations and then go create the YAML file. What about creating a R2D2 YAML as an intermediate if you wish to go that route?

Reply:
Just to clarify @guillaumevernieres I am okay with this for the short term, but I think we need a more elegant solution for all components eventually (or frankly, fix JEDI to not error if the obs are missing...)